dataset/src/test/scala/frameless/functions/UdfTest.scala

```diff
@@ -0,0 +1,163 @@
+package frameless
+package functions
+
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+
+class UdfTest extends TypedDatasetSuite {
+
+  test("one argument udf") {
+    def prop[A: TypedEncoder, B: TypedEncoder, C: TypedEncoder]
+    (data: Vector[X3[A, B, C]], f1: A => A): Prop = {
+      val dataset = TypedDataset.create(data)
+      val u1 = udf[X3[A, B, C], A, A](f1)
+      val u2 = dataset.makeUDF(f1)
+      val A = dataset.col[A]('a)
+
+      val dataset21 = dataset.select(u1(A)).collect().run().toVector
+      val dataset22 = dataset.select(u2(A)).collect().run().toVector
+      val d = data.map(x => f1(x.a))
+
+      (dataset21 ?= d) && (dataset22 ?= d)
+    }
+
+    check(forAll(prop[Int, Int, Int] _))
```

Comment:
I was thinking about this, there is a special case where `A` is case class (for instance `X2[Int, Int]`), and option, for instance `Option[Int]`